### PR TITLE
Dockerize python-threatexchange

### PR DESCRIPTION
**Pull request by dougneal**
Summary
---------

Reduce user friction by providing a Docker container for the CLI. This eliminates the need for the user to have to care about Python.

To make for a clean Docker image layout, the persistent state directory is now configurable. While the default remains `~/.threatexchange`, it can now be overridden by setting a `TX_STATEDIR` environment variable.

Test Plan
---------

```
$ docker build --tag threatexchange .
$ docker run threatexchange --help
usage: threatexchange [-h] [--factory-reset] [--version] [--verbose]
                      {config,fetch,match,label,dataset,hash} ...

Wrapper for the `threatexchange` library, can serve as a simple e2e solution.
...
```

### CODE REVIEW

**Review by BarrettOlson — COMMENTED** [review]
Awesome! here on in another PR I think it is worth updated the README.md with details/expected usage

**Comment by dougneal** [comment]
The README changes got stuck on my machine. Fixed

**Review by BarrettOlson — APPROVED** [review]

File: python-threatexchange/threatexchange/cli/main.py [hunk 1]
```diff
@@ -273,9 +273,9 @@ def _setup_logging(level_str: str, *, initial: bool = False) -> None:
 
 def inner_main(
     args: t.Optional[t.Sequence[t.Text]] = None,
-    state_dir: pathlib.Path = pathlib.Path("~/.threatexchange"),
```
Comment (Dcallies) [inline python-threatexchange/threatexchange/cli/main.py:276]: Why this change?

**Review by Dcallies — COMMENTED** [review]
I don't use a lot of docker - what does this allow that users might like?

File: python-threatexchange/threatexchange/cli/main.py (hunk 1 above)
Comment (dougneal) [reply python-threatexchange/threatexchange/cli/main.py:276]: This was a bad move. I thought it would be cleaner to move into the method body because it appeared unused. I hadn't considered the tests.

**Comment by dougneal** [comment]
> I don't use a lot of docker - what does this allow that users might like?

The value to the user is that they don't have to be Python developers to run it, or even care about their Python install at all. Just `docker build` followed by `docker run` and you have a functional CLI. The value will be even greater when you can spin up the API in the same manner.

If we were to publish the Docker image to the Docker hub or another container registry, they wouldn't even need the build step, just `docker run docker.example.com/threatexchange`.

File: python-threatexchange/threatexchange/cli/main.py (hunk 1 above)
Comment (BarrettOlson) [reply python-threatexchange/threatexchange/cli/main.py:276]: This seems reasonable, looks like the format/lint `black` is complaining though

**Review by Dcallies — APPROVED** [review]
So close! Just need to run black!
